script.js

```diff
@@ -0,0 +1,164 @@
+
+class RenderManager {
+    list = new ListManager();
+    submitButton = document.getElementById("addMission");
+    clearAllButton = document.getElementById("clear-all");
+    unFinishedConteiner = document.getElementById("unfinishedMissions")
+    finishedConteiner = document.getElementById("finishedMissions")
+
+    constructor() {
+        this.submitButton.addEventListener("click", this.handleNewItem.bind(this));
+        this.clearAllButton.addEventListener("click", () => {this.handleClearAll();});
+    }
+
+    render() {
+        this.finishedConteiner.innerHTML = "";
+        this.unFinishedConteiner.innerHTML = "";
+        for(const miss of this.list.missions){
+            const element = this.createMissionElement(miss);
+            if(miss.is_finished){
+                this.addMissionToFinished(element);
+            }else{
+                this.addMissionToUnfinished(element);
+            }
+        }
+    }
+ 
+    handleClearAll() {
+        const confirmation = confirm("Are you sure about it?");
+        if(confirmation){
+            this.clearAllLists();
+        }
+    }
+
+    clearAllLists() {
+        this.finishedConteiner.innerHTML = "";
+        this.unFinishedConteiner.innerHTML = "";
+        this.list = [];
```

Comment:
there is a bug here, can you find it?
regardless we prefer to handle all todos changes through the manager, and all html changes through the render method

Reply:
yes i got it.
change it in my code now